### PR TITLE
New module: xenserver_guest_facts - returns facts of XenServer VMs

### PR DESCRIPTION
**Pull request by bvitnik**
##### SUMMARY
This module implements fact gathering of XenServer VMs (guests) with module interface comparable to vmware_guest_facts module. In simple use cases, these modules are interchangeable. It complements xenserver_guest module.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
xenserver_guest_facts

##### ADDITIONAL INFORMATION
The module is trivially implemented since all the functionality is contained in xenserver module_util.

### CODE REVIEW

**Comment by Akasurde** [comment]
Same like other module, could you please add unit test cases for this change - [here](https://github.com/ansible/ansible/tree/devel/test/units/modules/cloud) ? Thanks.

**Comment by bvitnik** [comment]
@Akasurde Hi. Sorry for the delay. I've added some unit tests for the module. I will need this PR merged first before I commit unit tests for other module (xenserver_guest_powerstate, PR #49425) since these two modules will share some testing code.

As this module is just using functions from the xenserver module util, I've also implemented a complete suite of unit tests for xenserver module util itself in separate PR #50850 .

**Comment by bvitnik** [comment]
@Akasurde Hi. PR #50850 has been reviewed and merged by mattclay and I've also made appropriate changes to unit testing code in this PR. Can you please review this PR again and notify me if other changes are needed before the merge?

File: lib/ansible/modules/cloud/xenserver/xenserver_guest_facts.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        },
+        "state": "poweredon",
+        "uuid": "e3c0b2d5-5f05-424e-479c-d3df8b3e7cda",
+        "xenstore_data": {
+            "vm-data": ""
+        }
+    }
+'''
+
+import re
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/xenserver/xenserver_guest_facts.py]: Unused import.

File: lib/ansible/modules/cloud/xenserver/xenserver_guest_facts.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+HAS_XENAPI = False
+try:
+    import XenAPI
+    HAS_XENAPI = True
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_text, to_native
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/xenserver/xenserver_guest_facts.py]: Unused import

File: lib/ansible/modules/cloud/xenserver/xenserver_guest_facts.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+from ansible.module_utils import six
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/xenserver/xenserver_guest_facts.py]: Unused import.

**Review by Akasurde — CHANGES_REQUESTED** [review]
Rest LGTM

**Comment by bvitnik** [comment]
@Akasurde Done. Thanks for noticing unused imports.

**Review by Akasurde — APPROVED** [review]
LGTM

**Pull request merged**